src/Connection/index.js

```diff
@@ -588,7 +588,8 @@ export default class Connection extends Emitter {
       debug('creating socket connection on %s url', url)
     }
 
-    this.ws = new window.WebSocket(url)
+    // eslint-disable-next-line no-undef
+    this.ws = (typeof window !== 'undefined') ? new window.WebSocket(url) : new WebSocket(url)
```

Comment:
Can we move this check at the top level and then re-use it everywhere?

```js
const hasWindow = typeof window !== 'undefined'

hasWindow ? new window.Websocket(url) : WebSocket(url)
```

Reply:
Here you go